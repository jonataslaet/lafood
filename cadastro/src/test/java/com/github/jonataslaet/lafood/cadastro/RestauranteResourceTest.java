package com.github.jonataslaet.lafood.cadastro;

import static io.restassured.RestAssured.given;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;

@QuarkusTest
@DBRider
@DBUnit(schema = "public", cacheConnection = false, caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTestResource(CadastroTestLifeCycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-01.yml")
    public void recuperarTodosOsRestaurantes() {
        String resultado = given()
          .when().get("/restaurantes")
          .then()
             .statusCode(200)
             .extract().asString();
        Approvals.verify(resultado);
    }

}