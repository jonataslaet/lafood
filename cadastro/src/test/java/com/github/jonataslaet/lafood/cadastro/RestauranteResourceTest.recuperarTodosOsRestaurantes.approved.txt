[{"cnpj":"05510256389","id":123,"nome":"Novo Restaurante 01","proprietario":"ID do keycloak 01"},{"cnpj":"05510256389","id":321,"nome":"Novo Restaurante 02","proprietario":"ID do keycloak 02"}]